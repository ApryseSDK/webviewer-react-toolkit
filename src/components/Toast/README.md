This is a configurable Toast, but it has no inherent animations or timers. In
order to use the Toast as a notification system, you should use the `useToast`
hook.
