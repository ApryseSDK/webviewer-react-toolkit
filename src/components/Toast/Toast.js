'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.Toast = void 0;
var tslib_1 = require('tslib');
var classnames_1 = tslib_1.__importDefault(require('classnames'));
var react_1 = tslib_1.__importStar(require('react'));
var utils_1 = require('../../utils');
var Button_1 = require('../Button');
var Icon_1 = require('../Icon');
var IconButton_1 = require('../IconButton');
var Spinner_1 = require('../Spinner');
var Toast = function(_a) {
  var heading = _a.heading,
    children = _a.children,
    _b = _a.message,
    message = _b === void 0 ? 'info' : _b,
    onClose = _a.onClose,
    _c = _a.closeLabel,
    closeLabel = _c === void 0 ? 'Close' : _c,
    action = _a.action,
    className = _a.className,
    _d = _a.role,
    role = _d === void 0 ? 'status' : _d,
    _e = _a['aria-live'],
    ariaLive = _e === void 0 ? 'polite' : _e,
    _f = _a['aria-atomic'],
    ariaAtomic = _f === void 0 ? true : _f,
    props = tslib_1.__rest(_a, [
      'heading',
      'children',
      'message',
      'onClose',
      'closeLabel',
      'action',
      'className',
      'role',
      'aria-live',
      'aria-atomic',
    ]);
  var headingId = (0, react_1.useMemo)(function() {
    return (0, utils_1.getStringId)('toast_heading');
  }, []);
  var bodyId = (0, react_1.useMemo)(function() {
    return (0, utils_1.getStringId)('toast_body');
  }, []);
  var icon = (0, react_1.useMemo)(
    function() {
      switch (message) {
        case 'info':
          return 'Info';
        case 'success':
          return 'Success';
        case 'warning':
          return 'Warning';
        case 'error':
          return 'Error';
        case 'loading':
          return undefined;
      }
    },
    [message],
  );
  var toastClass = (0, classnames_1.default)(
    'ui__base ui__toast',
    'ui__toast--message-'.concat(message),
    className,
  );
  return react_1.default.createElement(
    'div',
    tslib_1.__assign(
      {
        'aria-labelledby': heading ? headingId : undefined,
        'aria-describedby': children ? bodyId : undefined,
      },
      props,
      {
        className: toastClass,
        role: role,
        'aria-live': ariaLive,
        'aria-atomic': ariaAtomic,
      },
    ),
    react_1.default.createElement('div', { className: 'ui__toast__border' }),
    icon
      ? react_1.default.createElement(Icon_1.Icon, {
          icon: icon,
          className: 'ui__toast__icon',
        })
      : react_1.default.createElement(Spinner_1.Spinner, {
          className: 'ui__toast__spinner',
        }),
    react_1.default.createElement(
      'div',
      { className: 'ui__toast__copy' },
      heading
        ? react_1.default.createElement(
            'div',
            { className: 'ui__toast__copy__heading', id: headingId },
            heading,
          )
        : undefined,
      children
        ? react_1.default.createElement(
            'div',
            { className: 'ui__toast__copy__body', id: bodyId },
            children,
          )
        : undefined,
    ),
    action
      ? react_1.default.createElement(
          'div',
          { className: 'ui__toast__action' },
          react_1.default.createElement(
            Button_1.Button,
            {
              className: 'ui__toast__button',
              onClick: action.onClick,
              buttonStyle: 'borderless',
            },
            action.text,
          ),
        )
      : undefined,
    onClose
      ? react_1.default.createElement(
          'div',
          { className: 'ui__toast__action' },
          react_1.default.createElement(
            IconButton_1.IconButton,
            { onClick: onClose, 'aria-label': closeLabel },
            react_1.default.createElement(Icon_1.Icon, { icon: 'Close' }),
          ),
        )
      : undefined,
  );
};
exports.Toast = Toast;
//# sourceMappingURL=Toast.js.map
