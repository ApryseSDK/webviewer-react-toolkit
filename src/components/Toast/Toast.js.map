{"version":3,"file":"Toast.js","sourceRoot":"","sources":["Toast.tsx"],"names":[],"mappings":";;;;AAAA,kEAAoC;AACpC,qDAA8F;AAE9F,qCAA0C;AAC1C,oCAAmC;AACnC,gCAA+B;AAC/B,4CAA2C;AAC3C,sCAAqC;AAoB9B,IAAM,KAAK,GAAmB,UAAC,EAYrC;IAXC,IAAA,OAAO,aAAA,EACP,QAAQ,cAAA,EACR,eAAgB,EAAhB,OAAO,mBAAG,MAAM,KAAA,EAChB,OAAO,aAAA,EACP,kBAAoB,EAApB,UAAU,mBAAG,OAAO,KAAA,EACpB,MAAM,YAAA,EACN,SAAS,eAAA,EACT,YAAe,EAAf,IAAI,mBAAG,QAAQ,KAAA,EACf,oBAAgC,EAAnB,QAAQ,mBAAG,QAAQ,KAAA,EAChC,sBAAgC,EAAjB,UAAU,mBAAG,IAAI,KAAA,EAC7B,KAAK,sBAX4B,sHAYrC,CADS;IAER,IAAM,SAAS,GAAG,IAAA,eAAO,EAAC,cAAM,OAAA,IAAA,mBAAW,EAAC,eAAe,CAAC,EAA5B,CAA4B,EAAE,EAAE,CAAC,CAAC;IAClE,IAAM,MAAM,GAAG,IAAA,eAAO,EAAC,cAAM,OAAA,IAAA,mBAAW,EAAC,YAAY,CAAC,EAAzB,CAAyB,EAAE,EAAE,CAAC,CAAC;IAE5D,IAAM,IAAI,GAAG,IAAA,eAAO,EAAC;QACnB,QAAQ,OAAO,EAAE;YACf,KAAK,MAAM;gBACT,OAAO,MAAM,CAAC;YAChB,KAAK,SAAS;gBACZ,OAAO,SAAS,CAAC;YACnB,KAAK,SAAS;gBACZ,OAAO,SAAS,CAAC;YACnB,KAAK,OAAO;gBACV,OAAO,OAAO,CAAC;YACjB,KAAK,SAAS;gBACZ,OAAO,SAAS,CAAC;SACpB;IACH,CAAC,EAAE,CAAC,OAAO,CAAC,CAAC,CAAC;IAEd,IAAM,UAAU,GAAG,IAAA,oBAAU,EAAC,oBAAoB,EAAE,6BAAsB,OAAO,CAAE,EAAE,SAAS,CAAC,CAAC;IAEhG,OAAO,CACL,2EACmB,OAAO,CAAC,CAAC,CAAC,SAAS,CAAC,CAAC,CAAC,SAAS,sBAC9B,QAAQ,CAAC,CAAC,CAAC,MAAM,CAAC,CAAC,CAAC,SAAS,IAC3C,KAAK,IACT,SAAS,EAAE,UAAU,EACrB,IAAI,EAAE,IAAI,eACC,QAAQ,iBACN,UAAU;QAEvB,uCAAK,SAAS,EAAC,mBAAmB,GAAG;QACpC,IAAI,CAAC,CAAC,CAAC,8BAAC,WAAI,IAAC,IAAI,EAAE,IAAI,EAAE,SAAS,EAAC,iBAAiB,GAAG,CAAC,CAAC,CAAC,8BAAC,iBAAO,IAAC,SAAS,EAAC,oBAAoB,GAAG;QACrG,uCAAK,SAAS,EAAC,iBAAiB;YAC7B,OAAO,CAAC,CAAC,CAAC,CACT,uCAAK,SAAS,EAAC,0BAA0B,EAAC,EAAE,EAAE,SAAS,IACpD,OAAO,CACJ,CACP,CAAC,CAAC,CAAC,SAAS;YACZ,QAAQ,CAAC,CAAC,CAAC,CACV,uCAAK,SAAS,EAAC,uBAAuB,EAAC,EAAE,EAAE,MAAM,IAC9C,QAAQ,CACL,CACP,CAAC,CAAC,CAAC,SAAS,CACT;QACL,MAAM,CAAC,CAAC,CAAC,CACR,uCAAK,SAAS,EAAC,mBAAmB;YAChC,8BAAC,eAAM,IAAC,SAAS,EAAC,mBAAmB,EAAC,OAAO,EAAE,MAAM,CAAC,OAAO,EAAE,WAAW,EAAC,YAAY,IACpF,MAAM,CAAC,IAAI,CACL,CACL,CACP,CAAC,CAAC,CAAC,SAAS;QACZ,OAAO,CAAC,CAAC,CAAC,CACT,uCAAK,SAAS,EAAC,mBAAmB;YAChC,8BAAC,uBAAU,IAAC,OAAO,EAAE,OAAO,gBAAc,UAAU;gBAClD,8BAAC,WAAI,IAAC,IAAI,EAAC,OAAO,GAAG,CACV,CACT,CACP,CAAC,CAAC,CAAC,SAAS,CACT,CACP,CAAC;AACJ,CAAC,CAAC;AAzEW,QAAA,KAAK,SAyEhB"}