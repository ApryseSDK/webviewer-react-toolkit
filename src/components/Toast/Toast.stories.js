'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.Basic = void 0;
var tslib_1 = require('tslib');
var addon_knobs_1 = require('@storybook/addon-knobs');
var react_1 = tslib_1.__importDefault(require('react'));
var action_1 = require('../../storybook-helpers/action/action');
var Toast_1 = require('../Toast');
var README_md_1 = tslib_1.__importDefault(require('./README.md'));
exports.default = {
  title: 'Components/Toast',
  component: Toast_1.Toast,
  parameters: { readme: README_md_1.default },
};
var Basic = function() {
  return react_1.default.createElement(
    'div',
    { style: { display: 'flex' } },
    react_1.default.createElement(
      Toast_1.Toast,
      {
        message: (0, addon_knobs_1.select)(
          'message',
          ['info', 'success', 'warning', 'error'],
          'info',
        ),
        heading: (0, addon_knobs_1.text)(
          'heading',
          'Information about the toast',
        ),
        onClose: (0, addon_knobs_1.boolean)('has onClose', false)
          ? (0, action_1.action)('onClose')
          : undefined,
        action: (0, addon_knobs_1.boolean)('has action', false)
          ? { text: 'Action', onClick: (0, action_1.action)('action.onClick') }
          : undefined,
      },
      (0, addon_knobs_1.text)('children', ''),
    ),
  );
};
exports.Basic = Basic;
//# sourceMappingURL=Toast.stories.js.map
