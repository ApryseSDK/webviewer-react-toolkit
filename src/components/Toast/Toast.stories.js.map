{"version":3,"file":"Toast.stories.js","sourceRoot":"","sources":["Toast.stories.tsx"],"names":[],"mappings":";;;;AAAA,sDAA+D;AAC/D,wDAA0B;AAC1B,gEAA+D;AAC/D,kCAAiC;AACjC,kEAAiC;AAEjC,kBAAe,EAAE,KAAK,EAAE,kBAAkB,EAAE,SAAS,EAAE,aAAK,EAAE,UAAU,EAAE,EAAE,MAAM,qBAAA,EAAE,EAAE,CAAC;AAEhF,IAAM,KAAK,GAAG,cAAM,OAAA,CACzB,uCAAK,KAAK,EAAE,EAAE,OAAO,EAAE,MAAM,EAAE;IAC7B,8BAAC,aAAK,IACJ,OAAO,EAAE,IAAA,oBAAM,EAAC,SAAS,EAAE,CAAC,MAAM,EAAE,SAAS,EAAE,SAAS,EAAE,OAAO,CAAC,EAAE,MAAM,CAAC,EAC3E,OAAO,EAAE,IAAA,kBAAI,EAAC,SAAS,EAAE,6BAA6B,CAAC,EACvD,OAAO,EAAE,IAAA,qBAAO,EAAC,aAAa,EAAE,KAAK,CAAC,CAAC,CAAC,CAAC,IAAA,eAAM,EAAC,SAAS,CAAC,CAAC,CAAC,CAAC,SAAS,EACtE,MAAM,EAAE,IAAA,qBAAO,EAAC,YAAY,EAAE,KAAK,CAAC,CAAC,CAAC,CAAC,EAAE,IAAI,EAAE,QAAQ,EAAE,OAAO,EAAE,IAAA,eAAM,EAAC,gBAAgB,CAAC,EAAE,CAAC,CAAC,CAAC,SAAS,IAEvG,IAAA,kBAAI,EAAC,UAAU,EAAE,EAAE,CAAC,CACf,CACJ,CACP,EAX0B,CAW1B,CAAC;AAXW,QAAA,KAAK,SAWhB"}