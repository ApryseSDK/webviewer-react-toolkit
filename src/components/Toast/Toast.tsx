import classnames from 'classnames';
import React, { AriaAttributes, FC, HTMLAttributes, MouseEventHandler, useMemo } from 'react';
import { CommonToastProps } from '../../hooks';
import { getStringId } from '../../utils';
import { Button } from '../Button';
import { Icon } from '../Icon';
import { IconButton } from '../IconButton';

export interface ToastProps extends CommonToastProps, HTMLAttributes<HTMLDivElement> {
  /**
   * If provided, toast will have a close button.
   */
  onClose?: MouseEventHandler<HTMLButtonElement>;
  /**
   * Provide alongside `onClose` for localized accessibility.
   * @default "Close"
   */
  closeLabel?: AriaAttributes['aria-label'];
  /** @default "status" */
  role?: HTMLAttributes<HTMLDivElement>['role'];
  /** @default "polite" */
  'aria-live'?: AriaAttributes['aria-live'];
  /** @default true */
  'aria-atomic'?: AriaAttributes['aria-atomic'];
}

export const Toast: FC<ToastProps> = ({
  heading,
  children,
  message = 'info',
  onClose,
  closeLabel = 'Close',
  action,
  className,
  role = 'status',
  'aria-live': ariaLive = 'polite',
  'aria-atomic': ariaAtomic = true,
  ...props
}) => {
  const headingId = useMemo(() => getStringId('toast_heading'), []);
  const bodyId = useMemo(() => getStringId('toast_body'), []);

  const icon = useMemo(() => {
    switch (message) {
      case 'info':
        return 'Info';
      case 'success':
        return 'Success';
      case 'warning':
        return 'Warning';
      case 'error':
        return 'Error';
    }
  }, [message]);

  const toastClass = classnames('ui__base ui__toast', `ui__toast--message-${message}`, className);

  return (
    <div
      aria-labelledby={heading ? headingId : undefined}
      aria-describedby={children ? bodyId : undefined}
      {...props}
      className={toastClass}
      role={role}
      aria-live={ariaLive}
      aria-atomic={ariaAtomic}
    >
      <div className="ui__toast__border" />
      <Icon icon={icon} className="ui__toast__icon" />
      <div className="ui__toast__copy">
        {heading ? (
          <div className="ui__toast__copy__heading" id={headingId}>
            {heading}
          </div>
        ) : (
          undefined
        )}
        {children ? (
          <div className="ui__toast__copy__body" id={bodyId}>
            {children}
          </div>
        ) : (
          undefined
        )}
      </div>
      {action ? (
        <div className="ui__toast__action">
          <Button className="ui__toast__button" onClick={action.onClick} buttonStyle="borderless">
            {action.text}
          </Button>
        </div>
      ) : (
        undefined
      )}
      {onClose ? (
        <div className="ui__toast__action">
          <IconButton onClick={onClose} aria-label={closeLabel}>
            <Icon icon="Close" />
          </IconButton>
        </div>
      ) : (
        undefined
      )}
    </div>
  );
};
