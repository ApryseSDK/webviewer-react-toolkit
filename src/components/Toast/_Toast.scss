.ui__toast {
  display: grid;
  display: flex;
  background-color: $color-gray-1;
  border-radius: $border-radius;
  overflow: hidden;
  box-shadow: $box-shadow-3;
  padding-right: 2px;

  @include for-phone-only {
    width: 100%;
  }

  @include for-tablet-up {
    max-width: 500px;
  }

  &--message {
    &-info {
      .ui__toast__border {
        background-color: $color-message-info;
      }
      .ui__toast__icon {
        fill: $color-message-info;
      }
    }

    &-warning {
      .ui__toast__border {
        background-color: $color-message-warning;
      }
      .ui__toast__icon {
        fill: $color-message-warning;
      }
    }

    &-error {
      .ui__toast__border {
        background-color: $color-message-error;
      }
      .ui__toast__icon {
        fill: $color-message-error;
      }
    }

    &-success {
      .ui__toast__border {
        background-color: $color-message-success;
      }
      .ui__toast__icon {
        fill: $color-message-success;
      }
    }
  }

  &__border {
    width: 4px;
    flex-shrink: 0;
  }

  &__icon {
    margin: 0 $padding-medium;
    @include flex-center;
    flex-shrink: 0;
  }

  &__copy {
    padding: $padding-small $padding $padding-small 0;
    display: flex;
    flex-direction: column;
    justify-content: center;
    flex: 1 1 auto;

    &__heading {
      flex-shrink: 0;
      color: $color-font-primary;
      font-weight: $font-weight-bold;
      padding: 2px 0;
    }

    &__body {
      flex-shrink: 0;
      color: $color-font-secondary;
      padding: 2px 0;
    }
  }

  &__action {
    display: flex;
    justify-content: flex-end;
    align-items: center;
    margin-left: $padding-tiny;
    flex-grow: 0;
  }

  &__button {
    width: auto;
    min-width: auto;
    padding: 0 $padding-tiny;
  }
}
