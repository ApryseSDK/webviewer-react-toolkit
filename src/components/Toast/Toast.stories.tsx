import { action } from '@storybook/addon-actions';
import { boolean, select, text } from '@storybook/addon-knobs';
import React from 'react';
import { Toast } from '../Toast';
import readme from './README.md';

export default { title: 'Components/Toast', component: Toast, parameters: { readme } };

export const Basic = () => (
  <div style={{ display: 'flex' }}>
    <Toast
      toastType={select('toastType', ['info', 'success', 'warning', 'error'], 'info')}
      heading={text('heading', 'Information about the toast')}
      onClose={boolean('has onClose', false) ? action('onClose') : undefined}
      action={boolean('has action', false) ? { text: 'Action', onClick: action('action.onClick') } : undefined}
    >
      {text('children', '')}
    </Toast>
  </div>
);
