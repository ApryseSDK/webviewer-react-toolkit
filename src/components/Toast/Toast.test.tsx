import { shallow } from 'enzyme';
import React from 'react';
import { Toast } from '../Toast';

jest.mock('../../utils', () => ({ getStringId: (prefix: string) => `${prefix}_1234` }));

describe('Toast component', () => {
  it('renders its contents', () => {
    const toast = shallow(<Toast heading={''} />);
    expect(toast.find('.ui__toast')).toHaveLength(1);
  });

  it('snapshot renders default toast', () => {
    const toast = shallow(<Toast heading={''} />);
    expect(toast).toMatchSnapshot();
  });
});
