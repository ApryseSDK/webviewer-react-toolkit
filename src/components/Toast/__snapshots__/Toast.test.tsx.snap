// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast component snapshot renders default toast 1`] = `
<div
  aria-atomic={true}
  aria-describedby="modal_body_1234"
  aria-labelledby="modal_heading_1234"
  aria-live="polite"
  className="ui__base ui__toast ui__toast--type-info"
  role="status"
>
  <div
    className="ui__toast__border"
  />
  <div
    className="ui__toast__icon"
  >
    <Info />
  </div>
  <div
    className="ui__toast__copy"
  />
</div>
`;
